pqt.so/src/action.cpp

```diff
@@ -402,7 +402,7 @@ int pqt_action::create_threads() {
   gpu_get_all_gpu_id(&gpu_id);
   gpu_get_all_device_id(&gpu_device_id);
 
-  for (size_t i = 0; i < gpu_id.size(); i++) {    // all possible sources
+  for (size_t i = 0; i < gpu_id.size()-1; i++) {    // all possible sources
```

Comment:
Since its <, for full iteration we need **gpu_id.size()**. Right ?

Reply:
Yeah this is src. I mean it should have atleast one destination to do transfer right? So i will be from 0 to n-1 and j will be from i+1 to n, thus we can cover all. i reaching n will result in no destinations